enhancements/host-network-configuration.md

```diff
@@ -0,0 +1,439 @@
+is expected to operate.  It would be wasteful to dedicate an entire NIC just
+for provisioning and no other purpose.  Instead, the administrator would like
+to bring up the cluster’s primary network as a VLAN on the same network
+interface that was used for provisioning.
+
+* Bare metal servers with a single network interface
+* We desire PXE booting these servers on the untagged network for provisioning
+* We must configure a VLAN on this interface to access the primary external
+  network for the cluster
+* Access to this network must be configured very early, before ignition will be
+  able to reach the machine config server
+
+### Use Case #2 - Extra VLAN, Post Install Configuration
+
+Consider an on-premise environment with workloads that must be able to access
+resources on multiple networks that must be directly connected to the cluster.
+One example could be an isolated network running database services that are not
+routeable from any other network.  Instead, we’ll attach the Nodes to that VLAN
+and limit which pods can access it.
+
+* After cluster installation, configure another VLAN on a network interface
+  that specific workloads will need access to
+
+### Use Case #3 - Static Network Configuration Only
+
+Consider a lab environment where DHCP services either aren’t available at all,
+or the person trying out an OpenShift cluster does not have access to it to
+configure DHCP address reservations.  As long as the cluster administrator has
+a pool of addresses they are allowed to assign manually, a cluster with fully
+static network configuration would allow them to install OpenShift without
+needing to work with the network administrator on DHCP server configuration.
+
+* Must be able to override default RHCOS behavior of requiring DHCP to be
+  present on at least one interface
+* Must be able to configure static network configuration for every interface
+  used by the cluster
+
+
+## Existing Configuration Interfaces
+
+### Summary
+
+First, here’s a summary of how an administrator may configure host networking
+in a cluster.  More details about each interface are in the following sections.
+
+#### First Boot
+
+When an RHCOS host first boots, the initramfs environment must be able to reach
+the [Machine Config Server
+(MCS)](https://github.com/openshift/machine-config-operator/blob/master/docs/MachineConfigServer.md)
+for Ignition to succeed.  The MCS starts by running on the Boostrap host while
+the first hosts are being deployed.  Ignition will attempt to reach the Machine
+Config Server via an included URL. By default, dracut will attempt to do DHCPv4
+and DHCPv6 on all interfaces and wait for one of them to work.  To change this
+behavior, an admin can:
+
+* Change the `ip=<...>` kernel argument(s) via PXE configuration or at the grub
+  command line during boot.  If doing a bare metal install, it is also possible
+  to edit the arguments baked into the image using a tool such as
+  `virt-customize`.
```

Comment:
This is true, though I don't know if we should encourage it's use as we may change the internals and this may become not true in the future.

Reply:
Yeah, agree with that.  My primary goal was to document the current reality for better or worse.  I'll add a note that this interface should be considered last resort.  Maybe that's good enough?  If we were adapting this into user facing documentation, we might choose to exclude some details.